frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart

```diff
@@ -143,18 +145,51 @@ class _GridPageState extends State<GridPage> {
           }
         },
         child: BlocConsumer<GridBloc, GridState>(
-          listener: (context, state) => state.loadingState.whenOrNull(
-            // If initial row id is defined, open row details overlay
-            finish: (_) {
-              if (widget.initialRowId != null && !_didOpenInitialRow) {
-                _didOpenInitialRow = true;
-
-                _openRow(context, widget.initialRowId!);
-              }
+          listener: (context, state) {
+            state.loadingState.whenOrNull(
```

Comment:
Abstract the anonymous callback into an onListen function

Reply:
Sounds good